{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "BLAST.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "authorship_tag": "ABX9TyMKeYLuKDxANfjSohxXsGrQ",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/pjd-code/BLAST/blob/main/BLAST.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "GS-NMJVm_VEv"
      },
      "source": [
        "# **B**asic **L**ocal **A**lignment **S**earch **T**ool"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ocqJEMZf_aN5"
      },
      "source": [
        "### We will start the module by setting up our Colab environment."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_g31gifhl3oU",
        "outputId": "32179c39-c705-4099-9f78-e537ee1eeee8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "source": [
        "%%capture\n",
        "#clean up the Colab environment\n",
        "!rm *.csv *.md *.json\n",
        "\n",
        "#change dirctory to /bin and retrive the BLAST executables\n",
        "%cd /bin\n",
        "!wget https://ftp.ncbi.nlm.nih.gov/blast/executables/LATEST/ncbi-blast-2.13.0+-x64-linux.tar.gz -P /bin/\n",
        "\n",
        "#unzip the file\n",
        "!tar zxvpf ncbi-blast-2.13.0+-x64-linux.tar.gz\n",
        "\n",
        "#add the blast executables to $PATH this path is specific to COLAB and will vary by operating system\n",
        "%env PATH=$PATH:/bin/ncbi-blast-2.13.0+/bin/:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin"
      ],
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "env: PATH=$PATH:/bin/ncbi-blast-2.13.0+/bin/:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "E2FmFG-iuJcU",
        "outputId": "3da8cdc1-ce8e-41f1-cda2-084361bad2c8"
      },
      "source": [
        "!git clone https://github.com/pjd-code/BLAST.git"
      ],
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fatal: destination path 'BLAST' already exists and is not an empty directory.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "QD4E6jGAzJ80",
        "outputId": "8052a9f7-d60c-437a-8079-16ef7e60fc15",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "source": [
        "#execue this block to test your setup\n",
        "!blastn -help"
      ],
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "USAGE\n",
            "  blastn [-h] [-help] [-import_search_strategy filename]\n",
            "    [-export_search_strategy filename] [-task task_name] [-db database_name]\n",
            "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\n",
            "    [-negative_gilist filename] [-negative_seqidlist filename]\n",
            "    [-taxids taxids] [-negative_taxids taxids] [-taxidlist filename]\n",
            "    [-negative_taxidlist filename] [-entrez_query entrez_query]\n",
            "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\n",
            "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\n",
            "    [-out output_file] [-evalue evalue] [-word_size int_value]\n",
            "    [-gapopen open_penalty] [-gapextend extend_penalty]\n",
            "    [-perc_identity float_value] [-qcov_hsp_perc float_value]\n",
            "    [-max_hsps int_value] [-xdrop_ungap float_value] [-xdrop_gap float_value]\n",
            "    [-xdrop_gap_final float_value] [-searchsp int_value]\n",
            "    [-sum_stats bool_value] [-penalty penalty] [-reward reward] [-no_greedy]\n",
            "    [-min_raw_gapped_score int_value] [-template_type type]\n",
            "    [-template_length int_value] [-dust DUST_options]\n",
            "    [-filtering_db filtering_database]\n",
            "    [-window_masker_taxid window_masker_taxid]\n",
            "    [-window_masker_db window_masker_db] [-soft_masking soft_masking]\n",
            "    [-ungapped] [-culling_limit int_value] [-best_hit_overhang float_value]\n",
            "    [-best_hit_score_edge float_value] [-subject_besthit]\n",
            "    [-window_size int_value] [-off_diagonal_range int_value]\n",
            "    [-use_index boolean] [-index_name string] [-lcase_masking]\n",
            "    [-query_loc range] [-strand strand] [-parse_deflines] [-outfmt format]\n",
            "    [-show_gis] [-num_descriptions int_value] [-num_alignments int_value]\n",
            "    [-line_length line_length] [-html] [-sorthits sort_hits]\n",
            "    [-sorthsps sort_hsps] [-max_target_seqs num_sequences]\n",
            "    [-num_threads int_value] [-mt_mode int_value] [-remote] [-version]\n",
            "\n",
            "DESCRIPTION\n",
            "   Nucleotide-Nucleotide BLAST 2.13.0+\n",
            "\n",
            "OPTIONAL ARGUMENTS\n",
            " -h\n",
            "   Print USAGE and DESCRIPTION;  ignore all other parameters\n",
            " -help\n",
            "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\n",
            " -version\n",
            "   Print version number;  ignore other arguments\n",
            "\n",
            " *** Input query options\n",
            " -query <File_In>\n",
            "   Input file name\n",
            "   Default = `-'\n",
            " -query_loc <String>\n",
            "   Location on the query sequence in 1-based offsets (Format: start-stop)\n",
            " -strand <String, `both', `minus', `plus'>\n",
            "   Query strand(s) to search against database/subject\n",
            "   Default = `both'\n",
            "\n",
            " *** General search options\n",
            " -task <String, Permissible values: 'blastn' 'blastn-short' 'dc-megablast'\n",
            "                'megablast' 'rmblastn' >\n",
            "   Task to execute\n",
            "   Default = `megablast'\n",
            " -db <String>\n",
            "   BLAST database name\n",
            "    * Incompatible with:  subject, subject_loc\n",
            " -out <File_Out, file name length < 256>\n",
            "   Output file name\n",
            "   Default = `-'\n",
            " -evalue <Real>\n",
            "   Expectation value (E) threshold for saving hits \n",
            "   Default = `10'\n",
            " -word_size <Integer, >=4>\n",
            "   Word size for wordfinder algorithm (length of best perfect match)\n",
            " -gapopen <Integer>\n",
            "   Cost to open a gap\n",
            " -gapextend <Integer>\n",
            "   Cost to extend a gap\n",
            " -penalty <Integer, <=0>\n",
            "   Penalty for a nucleotide mismatch\n",
            " -reward <Integer, >=0>\n",
            "   Reward for a nucleotide match\n",
            " -use_index <Boolean>\n",
            "   Use MegaBLAST database index\n",
            "   Default = `false'\n",
            " -index_name <String>\n",
            "   MegaBLAST database index name (deprecated; use only for old style indices)\n",
            "\n",
            " *** BLAST-2-Sequences options\n",
            " -subject <File_In>\n",
            "   Subject sequence(s) to search\n",
            "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
            "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\n",
            "   db_soft_mask, db_hard_mask\n",
            " -subject_loc <String>\n",
            "   Location on the subject sequence in 1-based offsets (Format: start-stop)\n",
            "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
            "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\n",
            "   db_soft_mask, db_hard_mask, remote\n",
            "\n",
            " *** Formatting options\n",
            " -outfmt <String>\n",
            "   alignment view options:\n",
            "     0 = Pairwise,\n",
            "     1 = Query-anchored showing identities,\n",
            "     2 = Query-anchored no identities,\n",
            "     3 = Flat query-anchored showing identities,\n",
            "     4 = Flat query-anchored no identities,\n",
            "     5 = BLAST XML,\n",
            "     6 = Tabular,\n",
            "     7 = Tabular with comment lines,\n",
            "     8 = Seqalign (Text ASN.1),\n",
            "     9 = Seqalign (Binary ASN.1),\n",
            "    10 = Comma-separated values,\n",
            "    11 = BLAST archive (ASN.1),\n",
            "    12 = Seqalign (JSON),\n",
            "    13 = Multiple-file BLAST JSON,\n",
            "    14 = Multiple-file BLAST XML2,\n",
            "    15 = Single-file BLAST JSON,\n",
            "    16 = Single-file BLAST XML2,\n",
            "    17 = Sequence Alignment/Map (SAM),\n",
            "    18 = Organism Report\n",
            "   \n",
            "   Options 6, 7, 10 and 17 can be additionally configured to produce\n",
            "   a custom format specified by space delimited format specifiers,\n",
            "   or in the case of options 6, 7, and 10, by a token specified\n",
            "   by the delim keyword. E.g.: \"17 delim=@ qacc sacc score\".\n",
            "   The delim keyword must appear after the numeric output format\n",
            "   specification.\n",
            "   The supported format specifiers for options 6, 7 and 10 are:\n",
            "   \t    qseqid means Query Seq-id\n",
            "   \t       qgi means Query GI\n",
            "   \t      qacc means Query accesion\n",
            "   \t   qaccver means Query accesion.version\n",
            "   \t      qlen means Query sequence length\n",
            "   \t    sseqid means Subject Seq-id\n",
            "   \t sallseqid means All subject Seq-id(s), separated by a ';'\n",
            "   \t       sgi means Subject GI\n",
            "   \t    sallgi means All subject GIs\n",
            "   \t      sacc means Subject accession\n",
            "   \t   saccver means Subject accession.version\n",
            "   \t   sallacc means All subject accessions\n",
            "   \t      slen means Subject sequence length\n",
            "   \t    qstart means Start of alignment in query\n",
            "   \t      qend means End of alignment in query\n",
            "   \t    sstart means Start of alignment in subject\n",
            "   \t      send means End of alignment in subject\n",
            "   \t      qseq means Aligned part of query sequence\n",
            "   \t      sseq means Aligned part of subject sequence\n",
            "   \t    evalue means Expect value\n",
            "   \t  bitscore means Bit score\n",
            "   \t     score means Raw score\n",
            "   \t    length means Alignment length\n",
            "   \t    pident means Percentage of identical matches\n",
            "   \t    nident means Number of identical matches\n",
            "   \t  mismatch means Number of mismatches\n",
            "   \t  positive means Number of positive-scoring matches\n",
            "   \t   gapopen means Number of gap openings\n",
            "   \t      gaps means Total number of gaps\n",
            "   \t      ppos means Percentage of positive-scoring matches\n",
            "   \t    frames means Query and subject frames separated by a '/'\n",
            "   \t    qframe means Query frame\n",
            "   \t    sframe means Subject frame\n",
            "   \t      btop means Blast traceback operations (BTOP)\n",
            "   \t    staxid means Subject Taxonomy ID\n",
            "   \t  ssciname means Subject Scientific Name\n",
            "   \t  scomname means Subject Common Name\n",
            "   \tsblastname means Subject Blast Name\n",
            "   \t sskingdom means Subject Super Kingdom\n",
            "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\n",
            "   \t\t\t (in numerical order)\n",
            "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\n",
            "   \t scomnames means unique Subject Common Name(s), separated by a ';'\n",
            "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\n",
            "   \t\t\t (in alphabetical order)\n",
            "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\n",
            "   \t\t\t (in alphabetical order) \n",
            "   \t    stitle means Subject Title\n",
            "   \tsalltitles means All Subject Title(s), separated by a '<>'\n",
            "   \t   sstrand means Subject Strand\n",
            "   \t     qcovs means Query Coverage Per Subject\n",
            "   \t   qcovhsp means Query Coverage Per HSP\n",
            "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\n",
            "   When not provided, the default value is:\n",
            "   'qaccver saccver pident length mismatch gapopen qstart qend sstart send\n",
            "   evalue bitscore', which is equivalent to the keyword 'std'\n",
            "   The supported format specifier for option 17 is:\n",
            "   \t        SQ means Include Sequence Data\n",
            "   \t        SR means Subject as Reference Seq\n",
            "   Default = `0'\n",
            " -show_gis\n",
            "   Show NCBI GIs in deflines?\n",
            " -num_descriptions <Integer, >=0>\n",
            "   Number of database sequences to show one-line descriptions for\n",
            "   Not applicable for outfmt > 4\n",
            "   Default = `500'\n",
            "    * Incompatible with:  max_target_seqs\n",
            " -num_alignments <Integer, >=0>\n",
            "   Number of database sequences to show alignments for\n",
            "   Default = `250'\n",
            "    * Incompatible with:  max_target_seqs\n",
            " -line_length <Integer, >=1>\n",
            "   Line length for formatting alignments\n",
            "   Not applicable for outfmt > 4\n",
            "   Default = `60'\n",
            " -html\n",
            "   Produce HTML output?\n",
            " -sorthits <Integer, (>=0 and =<4)>\n",
            "   Sorting option for hits:\n",
            "   alignment view options:\n",
            "     0 = Sort by evalue,\n",
            "     1 = Sort by bit score,\n",
            "     2 = Sort by total score,\n",
            "     3 = Sort by percent identity,\n",
            "     4 = Sort by query coverage\n",
            "   Not applicable for outfmt > 4\n",
            " -sorthsps <Integer, (>=0 and =<4)>\n",
            "   Sorting option for hps:\n",
            "     0 = Sort by hsp evalue,\n",
            "     1 = Sort by hsp score,\n",
            "     2 = Sort by hsp query start,\n",
            "     3 = Sort by hsp percent identity,\n",
            "     4 = Sort by hsp subject start\n",
            "   Not applicable for outfmt != 0\n",
            "\n",
            " *** Query filtering options\n",
            " -dust <String>\n",
            "   Filter query sequence with DUST (Format: 'yes', 'level window linker', or\n",
            "   'no' to disable)\n",
            "   Default = `20 64 1'\n",
            " -filtering_db <String>\n",
            "   BLAST database containing filtering elements (i.e.: repeats)\n",
            " -window_masker_taxid <Integer>\n",
            "   Enable WindowMasker filtering using a Taxonomic ID\n",
            " -window_masker_db <String>\n",
            "   Enable WindowMasker filtering using this repeats database.\n",
            " -soft_masking <Boolean>\n",
            "   Apply filtering locations as soft masks\n",
            "   Default = `true'\n",
            " -lcase_masking\n",
            "   Use lower case filtering in query and subject sequence(s)?\n",
            "\n",
            " *** Restrict search or results\n",
            " -gilist <String>\n",
            "   Restrict search of database to list of GIs\n",
            "    * Incompatible with:  seqidlist, taxids, taxidlist, negative_gilist,\n",
            "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -seqidlist <String>\n",
            "   Restrict search of database to list of SeqIDs\n",
            "    * Incompatible with:  gilist, taxids, taxidlist, negative_gilist,\n",
            "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -negative_gilist <String>\n",
            "   Restrict search of database to everything except the specified GIs\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
            "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -negative_seqidlist <String>\n",
            "   Restrict search of database to everything except the specified SeqIDs\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
            "   negative_gilist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -taxids <String>\n",
            "   Restrict search of database to include only the specified taxonomy IDs\n",
            "   (multiple IDs delimited by ',')\n",
            "    * Incompatible with:  gilist, seqidlist, taxidlist, negative_gilist,\n",
            "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -negative_taxids <String>\n",
            "   Restrict search of database to everything except the specified taxonomy IDs\n",
            "   (multiple IDs delimited by ',')\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
            "   negative_gilist, negative_seqidlist, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -taxidlist <String>\n",
            "   Restrict search of database to include only the specified taxonomy IDs\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, negative_gilist,\n",
            "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
            "   subject_loc\n",
            " -negative_taxidlist <String>\n",
            "   Restrict search of database to everything except the specified taxonomy IDs\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
            "   negative_gilist, negative_seqidlist, negative_taxids, remote, subject,\n",
            "   subject_loc\n",
            " -entrez_query <String>\n",
            "   Restrict search with the given Entrez query\n",
            "    * Requires:  remote\n",
            " -db_soft_mask <String>\n",
            "   Filtering algorithm ID to apply to the BLAST database as soft masking\n",
            "    * Incompatible with:  db_hard_mask, subject, subject_loc\n",
            " -db_hard_mask <String>\n",
            "   Filtering algorithm ID to apply to the BLAST database as hard masking\n",
            "    * Incompatible with:  db_soft_mask, subject, subject_loc\n",
            " -perc_identity <Real, 0..100>\n",
            "   Percent identity\n",
            " -qcov_hsp_perc <Real, 0..100>\n",
            "   Percent query coverage per hsp\n",
            " -max_hsps <Integer, >=1>\n",
            "   Set maximum number of HSPs per subject sequence to save for each query\n",
            " -culling_limit <Integer, >=0>\n",
            "   If the query range of a hit is enveloped by that of at least this many\n",
            "   higher-scoring hits, delete the hit\n",
            "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\n",
            " -best_hit_overhang <Real, (>0 and <0.5)>\n",
            "   Best Hit algorithm overhang value (recommended value: 0.1)\n",
            "    * Incompatible with:  culling_limit\n",
            " -best_hit_score_edge <Real, (>0 and <0.5)>\n",
            "   Best Hit algorithm score edge value (recommended value: 0.1)\n",
            "    * Incompatible with:  culling_limit\n",
            " -subject_besthit\n",
            "   Turn on best hit per subject sequence\n",
            " -max_target_seqs <Integer, >=1>\n",
            "   Maximum number of aligned sequences to keep \n",
            "   (value of 5 or more is recommended)\n",
            "   Default = `500'\n",
            "    * Incompatible with:  num_descriptions, num_alignments\n",
            "\n",
            " *** Discontiguous MegaBLAST options\n",
            " -template_type <String, `coding', `coding_and_optimal', `optimal'>\n",
            "   Discontiguous MegaBLAST template type\n",
            "    * Requires:  template_length\n",
            " -template_length <Integer, Permissible values: '16' '18' '21' >\n",
            "   Discontiguous MegaBLAST template length\n",
            "    * Requires:  template_type\n",
            "\n",
            " *** Statistical options\n",
            " -dbsize <Int8>\n",
            "   Effective length of the database \n",
            " -searchsp <Int8, >=0>\n",
            "   Effective length of the search space\n",
            " -sum_stats <Boolean>\n",
            "   Use sum statistics\n",
            "\n",
            " *** Search strategy options\n",
            " -import_search_strategy <File_In>\n",
            "   Search strategy to use\n",
            "    * Incompatible with:  export_search_strategy\n",
            " -export_search_strategy <File_Out>\n",
            "   File name to record the search strategy used\n",
            "    * Incompatible with:  import_search_strategy\n",
            "\n",
            " *** Extension options\n",
            " -xdrop_ungap <Real>\n",
            "   X-dropoff value (in bits) for ungapped extensions\n",
            " -xdrop_gap <Real>\n",
            "   X-dropoff value (in bits) for preliminary gapped extensions\n",
            " -xdrop_gap_final <Real>\n",
            "   X-dropoff value (in bits) for final gapped alignment\n",
            " -no_greedy\n",
            "   Use non-greedy dynamic programming extension\n",
            " -min_raw_gapped_score <Integer>\n",
            "   Minimum raw gapped score to keep an alignment in the preliminary gapped and\n",
            "   traceback stages\n",
            " -ungapped\n",
            "   Perform ungapped alignment only?\n",
            " -window_size <Integer, >=0>\n",
            "   Multiple hits window size, use 0 to specify 1-hit algorithm\n",
            " -off_diagonal_range <Integer, >=0>\n",
            "   Number of off-diagonals to search for the 2nd hit, use 0 to turn off\n",
            "   Default = `0'\n",
            "\n",
            " *** Miscellaneous options\n",
            " -parse_deflines\n",
            "   Should the query and subject defline(s) be parsed?\n",
            " -num_threads <Integer, >=1>\n",
            "   Number of threads (CPUs) to use in the BLAST search\n",
            "   Default = `1'\n",
            "    * Incompatible with:  remote\n",
            " -mt_mode <Integer, (>=0 and =<1)>\n",
            "   Multi-thread mode to use in BLAST search:\n",
            "    0 (auto) split by database \n",
            "    1 split by queries\n",
            "   Default = `0'\n",
            "    * Requires:  num_threads\n",
            " -remote\n",
            "   Execute search remotely?\n",
            "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
            "   negative_gilist, negative_seqidlist, negative_taxids, negative_taxidlist,\n",
            "   subject_loc, num_threads\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1C0T6VO-_yFg"
      },
      "source": [
        "Each Hight Scoring Pair is associated with a “bitscore” that is based on the similarity of the subsequences as determined by a particular set of rules. Because in larger subject sets some good matches are likely to be found by chance, each HSP is also associated with an “E value,” representing the expected number of matches one might find by chance in a subject set of that size with that score or better. For example, an E value of 0.05 means that we can expect a match by chance in 1 in 20 similar searches, whereas an E value of 2.0 means we can expect 2 matches by chance for each similar search."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "j3IrSF8kBJRQ"
      },
      "source": [
        "!blastn -db nt -query /content/BLAST/leptin_DNA.txt -remote -max_target_seqs 5"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iIr6C8zrx6c7"
      },
      "source": [
        "!blastp -db nr -query /bin/BLAST/leptin_protein.txt -remote -max_target_seqs 5"
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}